### DISCUSSION TITLE
REQUEST release appimage

### DISCUSSION THREAD

**Discussion by whysthatso** [solidtime-io/solidtime-desktop#93] [Feature Requests]
https://appimage.org/

**Comment by Onatcer** [comment]
Feel free to correct me if I misunderstood something here, I'm not too familiar with all the different distribution methods on Linux: 

We originally planned on doing AppImage releases, but we do rely on deeplinks that authenticate the application after the successful login. Those logins happen in a regular browser so the Application has to register the custom URL schema with the OS at some point. As far as I understand it, to do this with AppImage you need another application like AppImageLauncher installed so that the AppImage can register that custom URL schema. 

I did not get this to work without an additional dependency that has to be installed on the system, so I removed the builds from the release as I think this is very confusing.

**Comment by whysthatso** [comment]
that makes sense. you seem to have a better understanding of the implications already. 

so far i've been quite happy with the web version, so no need for me at the moment. besides not running in a browser tab, are there currently any conceivable feature differences to the web version?

**Comment by Onatcer** [comment]
it is currently quite similar, the desktop app has a floating window, menu bar controls etc. but there will be more features in the future that the web app does not have (AFK detection, activity tracking etc.)

what distro are you using? because i think the current builds should pretty much cover all popular distros...

**Comment by whysthatso** [comment]
i'm on archlinux, so i guess all it takes is to get some PKGBUILD up into AUR that takes the deb package. i'll put it onto my endless todo of 'nice to haves' in my infra, but i'm not sure i will get to it... ever :)

**Comment by Regen-forest** [comment]
Or maybe we can consider flatpak?

**Comment by stashymane** [comment]
+1 for Flatpak - Snaps are basically unheard of on distributions other than Ubuntu, and AppImages require you to distribute the application yourself, among other support issues.

**Comment by marcolaux** [comment]
Flatpak should be the way to go. Developers can target users directly, there is a store where they can be verified and users get trusted (and host agnostic) builds directly from the creators.

**Comment by Onatcer** [comment]
I'll move this into a discussion as we handle feature requests there so people can upvote it there. But i understand flatpak is the preferred option here and i'll look into that.

**Comment by Voronenko** [comment]
Well DEB versions are now having version link with glibc 2.38 via better_sqlite3.node addon

Uncaught Exception:
Error: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /opt/solidtime/resources/app.asar.unpacked/node_modules/better-sqlite3/build/Release/better_sqlite3.node)

which (glibc) is a usual pain...

which AppImage / Flatpack / Snap can generally can solve.
